huaweicloud/services/dli/resource_huaweicloud_dli_package.go

```diff
@@ -114,74 +92,60 @@ func buildDliDependentPackageCreateOpts(d *schema.ResourceData) resources.Create
-	}
-
-	// filter data by group name
-	filterData, err := utils.FilterSliceWithField(resList.Groups, map[string]interface{}{
-		"GroupName": d.Get("group_name").(string),
-	})
+	opt := buildDliDependentPackageCreateOpts(d)
+	resp, err := resources.CreateGroupAndUpload(c, opt)
 	if err != nil {
-		return diag.FromErr(err)
+		return diag.Errorf("error uploading package to OBS bucket: %s", err)
 	}
 
-	var resp *resources.Group
-	// If the group exists, upload the package to this group, otherwise create a new group before uploading the group.
-	if len(filterData) == 0 {
-		opt := buildDliDependentPackageCreateOpts(d)
-		resp, err = resources.CreateGroupAndUpload(c, opt)
-		if err != nil {
-			return diag.Errorf("error creating group and upload package: %s", err)
-		}
-	} else {
-		opt := buildDliDependentPackageUploadOpts(d)
-		resType := d.Get("type").(string)
+	// If the object list of resources.Resources is not empty, it means that the object has been uploaded successfully,
+	// and the element with an index of zero is the object name.
+	groupName := d.Get("group_name").(string)
 
-		resp, err = resources.Upload(c, uploadPath[resType], opt)
-		if err != nil {
-			return diag.Errorf("error uploading %s package to OBS bucket: %s", resType, err)
-		}
+	resourceNames := resp.Resources
+	if len(resourceNames) == 0 {
+		// Field resources.Group.ResourceNames is returned only when the group name is not specified.
+		resourceNames = resp.ResourceNames
 	}
 
-	// If the object list of resources.Resources is not empty, it means that the object has been uploaded successfully,
-	// and the element with an index of zero is the object name.
-	if len(resp.Resources) < 1 {
+	if len(resourceNames) < 1 {
 		return diag.Errorf("failed to upload package (%s).", d.Get("object_path").(string))
 	}
 	// object_path is not unique and cannot be used for ID setting, because the object can exist in multiple groups.
-	d.SetId(fmt.Sprintf("%s/%s", d.Get("group_name").(string), resp.Resources[0]))
+	if d.Get("group_name") != "" {
+		d.SetId(fmt.Sprintf("%s#%s", groupName, resp.Resources[0]))
+	} else {
+		d.SetId(resourceNames[0])
+	}
 
 	// If the owner of the configuration is not the creator, update it.
-	pkg, err := GetDliDependentPackageInfo(c, d.Id())
+	pkg, err := GetDliDependentPackageInfo(c, d.Id(), d.Get("group_name").(string))
 	if err != nil {
 		return diag.Errorf("error getting the package: %s", err)
 	}
+
 	if owner, ok := d.GetOk("owner"); ok && owner.(string) != pkg.Owner {
-		return ResourceDliDependentPackageV2Update(ctx, d, meta)
+		if err = updateOwner(c, d, pkg.ResourceName); err != nil {
+			return diag.FromErr(err)
+		}
 	}
 
 	return ResourceDliDependentPackageV2Read(ctx, d, meta)
 }
 
-func getGroupNameAndPackageName(id string) (groupName, packageName string, err error) {
-	names := strings.Split(id, "/")
+func getGroupNameAndPackageName(id, group string) (groupName, packageName string, err error) {
+	if group == "" {
+		return "", id, nil
+	}
```

Comment:
using split() function and if the length of result is equals one, just set the package name and return. If the the length is greater than one, set the element which the index is zero to the group name and set the element which the index is one to the package name.

Reply:
fixed